api/v1alpha4/conversion_test.go

```diff
@@ -244,6 +244,7 @@ func TestFuzzyConversion(t *testing.T) {
 						v1alpha5PortOpts.Network = nil
 					}
 				}
+				v1alpha5PortOpts.SecurityGroupFilters = nil
```

Comment:
@bavarianbidi is introducing `utilconversion.(Un)MarshalData` from Cluster API in #1247 to avoid losing data on a round-trip conversion to v1alpha4 and back to alpha5.

https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1247/files#diff-9a53c6afb893cfd6bb4f50185dea035075cb0ef25fce713667c270d5c9668a31R41

I think it makes sense to also apply this here, though I am not sure about the best process for this, some options:

- wait until #1247 is merged and then rebase ontop
- extract the round-trip setup into a seperate PR and rebase this ontop
- continue as is and do this in a follow up PR once #1247  is merged

Reply:
I think doing it in a follow up PR could be good as I am planning the deprecate the `securityGroups` field. The reason being, the uuid it provides is also provided by the new introduced `securityGroupFilter` field.

Reply:
I'm also fine doing a separate PR as it seems that all conversation unit tests has to be refactored. Already though about but didn't had much time to start the discussion.